pkg/events/event_ringbuffer.go

```diff
@@ -199,3 +199,41 @@ func newEventRingBuffer(capacity uint64) *eventRingBuffer {
 		events:   make([]*si.EventRecord, capacity),
 	}
 }
+
+// resize the existing ring buffer
+// this method will be called upon configuration reload
+func (e *eventRingBuffer) Resize(newSize uint64) {
```

Comment:
This definitely needs a unit test, with all branches being covered.

Reply:
Unit tests added